var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -62,136 +64,184 @@ class PyNumpy(PythonPackage):
     variant('lapack', default=True, description='Build with LAPACK support')
 
     depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
+    depends_on('python@2.7:2.8,3.5:', type=('build', 'run'), when='@1.16:')
+    depends_on('python@3.5:', type=('build', 'run'), when='@1.17:')
     depends_on('py-setuptools', type='build')
     depends_on('blas',   when='+blas')
     depends_on('lapack', when='+lapack')
 
     depends_on('py-nose@1.0.0:', when='@:1.14', type='test')
     depends_on('py-pytest', when='@1.15:', type='test')
 
-    def setup_dependent_package(self, module, dependent_spec):
-        python_version = self.spec['python'].version.up_to(2)
+    # Allows you to specify order of BLAS/LAPACK preference
+    # https://github.com/numpy/numpy/pull/13132
+    patch('blas-lapack-order.patch', when='@1.15:1.16')
 
-        self.spec.include = join_path(
-            self.prefix.lib,
-            'python{0}'.format(python_version),
-            'site-packages',
-            'numpy/core/include')
+    # GCC 4.8 is the minimum version that works
+    conflicts('%gcc@:4.7', msg='GCC 4.8+ required')
 
-    def patch(self):
+    def flag_handler(self, name, flags):
+        # -std=c99 at least required, old versions of GCC default to -std=c90
+        if self.spec.satisfies('%gcc@:5.1'):
+            flags.append(self.compiler.c99_flag)
```

Comment:
Hmm, this doesn't seem to work:
```
/usr/bin/ld: cannot find -l-std=c99
```
Any ideas? Should I just set `CFLAGS`?

Reply:
According to https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags this should work. I'll create an issue for it. That being said you may only need to do this for `cflags`, for example:

```
        if self.spec.satisfies('%gcc@:5.1') and name == 'cflags':
            flags.append(self.compiler.c99_flag)
```

does that work?

Reply:
This seems to be working! Thanks